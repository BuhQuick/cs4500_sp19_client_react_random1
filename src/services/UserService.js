export default class UserService {
    static instance = null;

    static getInstance() {
        if (UserService.instance === null) {
            UserService.instance = new UserService()
        }
        return this.instance
    }

    findUserById = userId =>
        fetch(MIDDLE_TIER_URL + "/api/users/${userId}")
            .then(response => response.json())
    findAllUsers = () =>
        fetch(MIDDLE_TIER_URL + "/api/users")
            .then(response => response.json())
    createUser = user =>
        fetch(MIDDLE_TIER_URL + "/api/users",
            {
                method: 'POST',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())
    updateUser = user =>
        fetch(MIDDLE_TIER_URL + "/api/users/${user.id},
            {
                method: 'PUT',
                body: JSON.stringify(user),
                headers: {'content-type': 'application/json'}
            })
            .then(response => response.json())
    deleteUser = user =>
        fetch(MIDDLE_TIER_URL + "/api/users/${user.id}",
            {
                method: 'DELETE',
            })
}