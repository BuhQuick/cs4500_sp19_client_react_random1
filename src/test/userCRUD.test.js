import React from 'react'
import TestRenderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import users from '../mock/data/users.mock.json'
import UserService from '../services/UserService'
import Users from "../components/Users";

const userService = UserService.getInstance();

test('Create and Delete Users', () => {
    let cool_ali = {id: 13, username: "coolali", password: "lolwutt", firstName: "Ali",
     lastName: "Gabri"};

    const saveUser = () => {
        users.push(cool_ali);

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

    };

    const deleteUser = userId => {
        users.splice(0, 1)

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

    };

    const testRenderer = TestRenderer.create(
        <Users users={users} editingUser={cool_ali} deleteUser={deleteUser} saveUser={saveUser}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
});

test('userRenders', () => {
    const testRenderer = TestRenderer.create(
        <Users users={users} editingUser={users[2]}/>
    );

    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();

    const testInstance = testRenderer.root;

    const usernameField = testInstance.findByProps({className: 'form-control username-edit'});
    const numUsers = testInstance.findAllByProps({className: 'btn btn-primary btn-block delete-user-btn'});

    expect(usernameField.props.value).toBe('Username');
    expect(numUsers.length).toBe(0);
});

test('userRendersService', () => {
    userService.findAllUser().then(users => {
        const testRenderer = TestRenderer.create(
            <Users users={users} editingUser={users[2]}/>
        );

        let tree = testRenderer.toJSON();
        expect(tree).toMatchSnapshot();

        const testInstance = testRenderer.root;

        const usernameField = testInstance.findByProps({className: 'form-control username-edit'});
        const numUsers = testInstance.findAllByProps({className: 'btn btn-primary btn-block delete-user-btn'});

        expect(usernameField.props.value).toBe('JamesWutang');
        expect(numUsers.length).toBe(users.length);
    });
});