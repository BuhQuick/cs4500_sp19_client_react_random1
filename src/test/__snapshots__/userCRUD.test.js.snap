// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create and Delete Users 1`] = `
<div>
  <h3>
    Users
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          First Name
        </th>
        <th
          scope="col"
        >
          Last Name
        </th>
        <th>
           
        </th>
      </tr>
      <tr>
        <th>
          <input
            className="form-control username-edit"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="Username"
          />
        </th>
        <th>
          <input
            className="form-control firstname-edit"
            onChange={[Function]}
            placeholder="firstName"
            type="text"
            value="First name"
          />
        </th>
        <th>
          <input
            className="form-control lastname-edit"
            onChange={[Function]}
            placeholder="lastName"
            type="text"
            value="Last name"
          />
        </th>
        <th>
          <button
            className="btn btn-primary btn-block create-btn-user"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
        </th>
        <th>
          <button
            className="btn btn-primary btn-block save-user-btn"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;

exports[`userRenders 1`] = `
<div>
  <h3>
    Users
  </h3>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          First Name
        </th>
        <th
          scope="col"
        >
          Last Name
        </th>
        <th>
           
        </th>
      </tr>
      <tr>
        <th>
          <input
            className="form-control username-edit"
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="Username"
          />
        </th>
        <th>
          <input
            className="form-control firstname-edit"
            onChange={[Function]}
            placeholder="firstName"
            type="text"
            value="First name"
          />
        </th>
        <th>
          <input
            className="form-control lastname-edit"
            onChange={[Function]}
            placeholder="lastName"
            type="text"
            value="Last name"
          />
        </th>
        <th>
          <button
            className="btn btn-primary btn-block create-btn-user"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
        </th>
        <th>
          <button
            className="btn btn-primary btn-block save-user-btn"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
